File: easy/234pylindrome_linked_list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def isPalindrome(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        newhead=None
        newnode=None
        prevnode=None
        node=head
        while node!=None:
        	newnode=ListNode(node.val)
        	if newhead==None:
        		newhead=newnode
        		prev=newnode
        	else:
        		prev.next=newnode
        		prev=newnode
        	node=node.next
        newhead=self.reverseList(newhead)
        node,newnode=head,newhead
        while node!=None:
        	if node.val==newnode.val:
        		node,newnode = node.next,newnode.next
        	else:
        		return False
        return True

    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head==None or head.next==None: return head
        prev,cur,nex=None,head,head.next
        while cur!=None:
        	cur.next=prev
        	prev=cur
        	cur=nex
        	nex=nex.next

File: easy/234pylindrome_linked_list.cpp
#include<iostream>
using namespace std;
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
 struct ListNode {
     int val;
     ListNode *next;
     ListNode(int x) : val(x), next(NULL) {}
 };
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        ListNode* prev,*node,*reverse;
        prev=NULL;
        node=head;
        while (node!=NULL){
            ListNode* newnode=new ListNode(node->val);
            newnode->next=prev;
            prev=newnode;
            node=node->next;
        }; 
        reverse=prev;
        ListNode* renode=reverse;
        node=head;
        while (node!=NULL){
            if (node->val!=renode->val){
                return false;
            }else{
                node=node->next;
                renode=renode->next;
            };
        };
        return true;
                
    }
};
int main(){
    cout<<123<<endl;
};
